### ISSUE TITLE
Using "here doc" syntax with datadog_monitor.message causes plan to always show changes

### ISSUE THREAD

**Issue by billputer** [hashicorp/terraform#5642]
With the datadog_monitor resource, I'm trying to set a multi-line message using the "here doc" syntax.

```
message = <<MESSAGE
The ${var.env} RDS instance is low on disk space.
Notify: ${var.notify}
MESSAGE
```

This applies cleanly, but it appears that Datadog trims trailing newlines, so that resource always shows as changed.

```
~ module.datadog.datadog_monitor.rds_disk
    message: "The staging RDS instance is low on disk space.\nNotify: @all" => "The staging RDS instance is low on disk space.\nNotify: @all\n"
```

Unfortunately, I can't find a clean way to set a multiline message without a trailing newline. I can work around this by either writing the message as a one-line string with "\n" inline, or by putting the message in a template, but both of those decrease readability. I'd like to be able to use multiline strings with the datadog_monitor resource.

**Comment by billputer** [comment]
My go-fu is poor, but I'll take a look at the provider, it might be fairly easy to have it trim newlines before comparing to the result from the Datadog API.

**Comment by billputer** [comment]
I was able to get the desired behavior by adding the following StateFunc.
```diff
diff --git a/builtin/providers/datadog/resource_datadog_monitor.go b/builtin/providers/datadog/resource_datadog_monitor.go
index af2b89f..79a28c6 100644
--- a/builtin/providers/datadog/resource_datadog_monitor.go
+++ b/builtin/providers/datadog/resource_datadog_monitor.go
@@ -27,6 +27,9 @@ func resourceDatadogMonitor() *schema.Resource {
                        "message": &schema.Schema{
                                Type:     schema.TypeString,
                                Required: true,
+                               StateFunc: func(val interface{}) string {
+                                       return strings.TrimSpace(val.(string))
+                               },
                        },
                        "escalation_message": &schema.Schema{
                                Type:     schema.TypeString,
```
I've also verified that the Datadog automatically trims leading and trailing white space, so adding this StateFunc should cause the message field to match. I'm going to check if any other fields on the datadog_monitor require normalization, and then put together a PR.

I'll also look around to see if there's any good examples of testing this type of StateFunc transformation, since I'm not really sure the canonical way to test this.

**Comment by ojongerius** [comment]
Awesome, thanks for taking this on!

**Comment by multilinear** [comment]
I ran in to the same issue, so I wrote a patch for it including a couple of other fields and tests. ^

**Comment by stack72** [comment]
Hi @billputer this has been merged in #5788 

Thanks for the report!

**Comment by ojongerius** [comment]
Thanks @stack72
